services/collaboration/src/controllers/roomController.ts

```diff
@@ -0,0 +1,186 @@
+import { Request, Response } from 'express';
+import {
+    createRoomInDB,
+    createYjsDocument,
+    deleteYjsDocument,
+    findRoomById,
+    findRoomsByUserId,
+    closeRoomById,
+    updateRoomUserStatus,
+} from '../services/mongodbService';
+import axios from 'axios';
+import { handleNotFound, handleSuccess, handleServerError, handleBadRequest } from '../utils/helper';
+import config from '../config';
+
+/**
+ * Create a room with users, question details, and Yjs document
+ * @param user1
+ * @param user2
+ * @param topics
+ * @param difficulty
+ * @returns roomId
+ */
+export const createRoomWithQuestion = async (user1: any, user2: any, topics: string[], difficulty: string) => {
+    try {
+        const questionServiceUrl = config.QUESTION_SERVICE_URL;
+        const topicString = topics.join(',');
+        const response = await axios.get(
+            `${questionServiceUrl}questions/search?topics=${topicString}&difficulty=${difficulty}&limit=1`,
+        );
+
+        if (response.data && response.data.data && response.data.data.length > 0) {
+            const question = response.data.data[0];
+
+            const roomData = {
+                users: [
+                    {
+                        ...user1,
+                        isForfeit: false,
+                    },
+                    {
+                        ...user2,
+                        isForfeit: false,
+                    },
+                ],
+                question_id: question.id,
+                createdAt: new Date(),
+            };
+
+            const room_id = await createRoomInDB(roomData);
+            console.log('Room created with ID:', room_id);
+
+            await createYjsDocument(room_id.toString());
```

Comment:
`createYjsDocument(room_id.toString())` is duplicated, it is used inside `createRoomInDb(roomData)` just above as well